product_import/wizard/product_import.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2022 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+import mimetypes
+from base64 import b64decode, b64encode
+from datetime import date, timedelta
+
+from lxml import etree
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+logger = logging.getLogger(__name__)
+
+
+class ProductImport(models.TransientModel):
+    _name = "product.import"
+    _description = "Product import from files"
+
+    product_file = fields.Binary(
+        string="Product Catalogue",
+        required=True,
+        help="Upload a Product Catalogue",
+    )
+    product_filename = fields.Char(string="Filename")
+
+    @property
+    def _bdimport(self):
+        return self.env["business.document.import"]
+
+    @api.onchange("product_file")
+    def product_file_change(self):
+        if not self.product_filename or not self.product_file:
+            return
+
+        doc_type = self._parse_file(
+            self.product_filename, b64decode(self.product_file), detect_doc_type=True
+        )
+        if doc_type is None:
+            return {"warning": self._unsupported_file_msg(self.product_filename)}
+
+    def _parse_file(self, filename, filecontent, detect_doc_type=False):
+        assert filename, "Missing filename"
+        assert filecontent, "Missing file content"
+        filetype = mimetypes.guess_type(filename)
+        logger.debug("File mimetype: %s", filetype)
+        mimetype = filetype[0]
+        supported_types = {
```

Comment:
pls drop this abandoned support for CSV, I know it comes from sale_order_import but it should be removed and refactored there too, thanks!